aerofiles/seeyou/reader.py

```diff
@@ -193,3 +193,90 @@ def decode_description(self, description):
             return None
 
         return description
+
+    def decode_taskname(self, fields):
+        if not fields:
+            raise ParserError('Name field must not be empty')
+
+        return fields[0]
+
+    def decode_tasklist(self, fields, waypoints):
```

Comment:
i have some difficulty understanding that part of the code. let me explain what i think that happens (please correct me when i'm wrong):

- the `next()` method provides the logic for the next "element" in the object (in the case of a cup file with only waypoints, this method gives the next waypoint). in the case of a cup file with waypoints and a task, this handling is not yet covered.
- the call to the methods `decode_tasklist()`, `decode_taskname`, `decode_options`, `decode_taskpoint` could be incorporated in some way into this `next()` method to ultimately provide the task back to the user.

the `__iter__()` method seems a bit counter intuitive if the object contains both waypoints and a task. what are your thoughts on this?